src/renderer/components/inputs/InputContainer.tsx

```diff
@@ -1,64 +1,208 @@
+            // Any other inputs should be invalid
+            if (connectingFrom.handleType === 'target') {
+                return false;
+            }
+            // Show same types
+            const connectionIsValid = isValidConnection({
+                source: connectingFrom.nodeId,
+                sourceHandle: connectingFrom.handleId,
+                target: id,
+                targetHandle: `${id}-${inputId}`,
+            });
+            if (connectionIsValid && intersect(connectingFromType, type).type !== 'never') {
+                return true;
+            }
+            return false;
+        }, [connectingFrom, connectingFromType, type, id, inputId]);
+
+        const { typeDefinitions, functionDefinitions } = useContext(GlobalContext);
+        const { useIsDarkMode } = useContext(SettingsContext);
+        const [isDarkMode] = useIsDarkMode;
 
         let contents = children;
-        const handleColor = useColorModeValue('#EDF2F7', '#171923');
-        const borderColor = useColorModeValue('#171923', '#F7FAFC');
+        const handleColors = getTypeAccentColors(type, typeDefinitions, isDarkMode); // useColorModeValue('#EDF2F7', '#171923');
+
+        const parentTypeColor = useMemo(() => {
+            if (connectedEdge) {
+                const parentNode: Node<NodeData> = getNode(connectedEdge.source)! as Node<NodeData>;
+                const parentInOutId = parseHandle(connectedEdge.sourceHandle!).inOutId;
+                const parentType = functionDefinitions
+                    .get(parentNode.data.schemaId)!
+                    .outputDefaults.get(parentInOutId)!;
+                return getTypeAccentColors(parentType, typeDefinitions, isDarkMode)[0];
+            }
+            return null;
+        }, [connectedEdge, typeDefinitions, functionDefinitions, getNode, isDarkMode]);
+
+        // A conic gradient that uses all handle colors to give an even distribution of colors
+        const handleColorString = handleColors
+            .map((color, index) => {
+                const percent = index / handleColors.length;
+                const nextPercent = (index + 1) / handleColors.length;
+                return `${color} ${percent * 100}% ${nextPercent * 100}%`;
+            })
+            .join(', ');
+        const handleGradient = `conic-gradient(from 90deg, ${handleColorString})`;
+        const connectedColor = useColorModeValue('#EDF2F7', '#171923');
         if (hasHandle) {
             contents = (
                 <HStack
                     h="full"
                     sx={{
                         '.react-flow__handle-connecting': {
-                            background: '#E53E3E !important',
-                        },
-                        '.react-flow__handle-valid': {
-                            background: '#38A169 !important',
+                            opacity: showHandle ? 1 : 0,
                         },
+                        '.react-flow__handle-valid': {},
```

Comment:
Does this even do anything?

Reply:
No, I just don't wanna have to go look at the docs to find that again if I wanna use it for something in the future

Reply:
Never mind, I can just get rid of this whole style prop